### PR TITLE
Crypto: minor refactoring

### PR DESCRIPTION
**Pull request by Nekotekina**
~~I didn't notice where it gets restored to zeros.
Maybe I'm wrong about this.~~

### CODE REVIEW

**Comment by Nekotekina** [comment]
Nvm, found it. Really weird design.

**Comment by Nekotekina** [comment]
Made global elliptic curve context thread_local, should be safe plus it's small.

**Comment by Nekotekina** [comment]
Restored what I had to revert, now double checked that const_cast is legal in this place.

File: rpcs3/Crypto/ec.cpp [hunk 1]
```diff
@@ -482,19 +482,19 @@ int ecdsa_set_curve(u8* p, u8* a, u8* b, u8* N, u8* Gx, u8* Gy)
...
 }
 
-void ecdsa_set_priv(u8 *k)
+void ecdsa_set_priv(const u8* k)
 {
 	memcpy(ec_k, k, sizeof ec_k);
 }
 
-int ecdsa_verify(u8 *hash, u8 *R, u8 *S)
+int ecdsa_verify(u8* hash, u8 *R, u8 *S)
```
Comment (Megamouse) [inline rpcs3/Crypto/ec.cpp:497]:
```suggestion
int ecdsa_verify(u8* hash, u8* R, u8* S)
```

Comment (Nekotekina) [reply rpcs3/Crypto/ec.cpp:497]: Haha okay

Comment (Nekotekina) [reply rpcs3/Crypto/ec.cpp]: Applied clang-format partially

**Pull request merged**